Chummer/Backend/Derived/SustainedSpell.cs

```diff
@@ -0,0 +1,193 @@
+/*  This file is part of Chummer5a.
+ *
+ *  Chummer5a is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Chummer5a is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Chummer5a.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  You can obtain the full source code for Chummer5a at
+ *  https://github.com/chummer5a/chummer5a
+ */
+
+using Chummer.Backend.Skills;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Windows.Forms;
+using System.Xml;
+using NLog;
+
+namespace Chummer
+{
+    /// <summary>
+    /// A Sustained Magician Spell
+    /// </summary>
+    [HubClassTag("SourceID", true, "Name", "Extra")]
+    [DebuggerDisplay("{DisplayName(GlobalOptions.DefaultLanguage)}")]
+
+    public class SustainedSpell : Spell, IHasInternalId, ISustainable, INotifyPropertyChanged
+    {
+        private Guid _guiID;
+        private Guid _guiSourceID = Guid.Empty;
+        private bool _blnSelfSustained = true;
+        private int _intForce = 0;
+        private int _intNetHits = 0;
+        private readonly Character _objCharacter;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        #region Constructor, Create, Save, Load, and Print Methods
+
+        public SustainedSpell(Character objCharacter) : base(objCharacter)
+        {
+            //Create the GUID for new sustained spells
+            _guiID = Guid.NewGuid();
+            _objCharacter = objCharacter;
+
+            PropertyChanged += OnSustainedChanged;
```

Comment:
Careful with this assignment. Because `OnSustinedChanged` refreshes character's sustaining penalties every time a property of this `SustainedSpell` is changed, things are going to go haywire whenever `CreateByMapping` is called,